### PR TITLE
Created Preorder & Postorder traversals in tree

### PR DESCRIPTION
**Pull request by lg970325**
I created two tree traversal algorithms - preorder & postorder. Each of them has two functions recursive and non-recursive. I added sample test in the same file, which is same as inorder traversal.

- [X] **If creating a new file :**
  - [X] added links to it in the README files ?
  - [X] included tests with it ?
  - [X] added description (overview of algorithm, time and space compleixty, and possible edge case) in docstrings ?

### CODE REVIEW

File: algorithms/tree/traversal/postorder.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    if res is None:
+        res = []
+    postorder_rec(root.left, res)
+    postorder_rec(root.right, res)
+    res.append(root.val)
+    return res
+
+
+
+if __name__ == '__main__':
```
Comment (goswami-rahul) [inline algorithms/tree/traversal/postorder.py]: This testing part should go to the `test` folder.

File: algorithms/tree/traversal/preorder.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+    if res is None:
+        res = []
+    res.append(root.val)
+    preorder_rec(root.left, res)
+    preorder_rec(root.right, res)
+    return res
+
+
+
+if __name__ == '__main__':
```
Comment (goswami-rahul) [inline algorithms/tree/traversal/preorder.py]: move to `test` folder

**Comment by danghai** [comment]
@lg970325 We are missing the unittest for `tree`. Do you want to do it?

**Comment by lg970325** [comment]
I created test_tree.py and moved test cases in postorder and preorder traversals. :)

**Comment by lg970325** [comment]
Created test_tree.py and moved test cases in preorder & postorder there :)

**Pull request merged**